assets/js/base/hocs/with-reviews.js

```diff
@@ -33,27 +31,50 @@ const withReviews = ( OriginalComponent ) => {
 
 		componentDidUpdate( prevProps ) {
 			if ( prevProps.reviewsToDisplay < this.props.reviewsToDisplay ) {
-				// Since this attribute is controlled with a slider,
-				// it's better not to load the reviews immediately.
+				// Since this attribute might be controlled via something with
+				// short intervals between value changes, this allows for optionally
+				// delaying review fetches via the provided delay function.
 				this.delayedAppendReviews();
 			} else if (
-				prevProps.categoryIds !== this.props.categoryIds ||
-				prevProps.orderby !== this.props.orderby ||
-				prevProps.order !== this.props.order ||
-				prevProps.productId !== this.props.productId
+				this.shouldReplaceReviews( prevProps, this.props )
 			) {
 				this.replaceReviews();
 			}
 		}
 
-		getArgs() {
-			const { categoryIds, order, orderby, productId } = this.props;
+		shouldReplaceReviews( prevProps, nextProps ) {
+			if (
+				prevProps.orderby !== nextProps.orderby ||
+				prevProps.order !== nextProps.order ||
+				prevProps.productId !== nextProps.productId
+			) {
+				return true;
+			}
+
+			if ( prevProps.categoryIds === nextProps.categoryIds ) {
+				return false;
+			}
+
+			const prevCategoryIds = Array.isArray( prevProps.categoryIds ) ?
+				prevProps.categoryIds.join( ',' ) :
+				prevProps.categoryIds;
+			const nextCategoryIds = Array.isArray( nextProps.categoryIds ) ?
+				nextProps.categoryIds.join( ',' ) :
+				nextProps.categoryIds;
+
+			return prevCategoryIds !== nextCategoryIds;
```

Comment:
All the category_id logic here would be much cleaner if you used `is-shallow-equal` :).  I saw your comment about not including it as a dependency but it's only [815B minified ](https://bundlephobia.com/result?p=@wordpress/is-shallow-equal@1.5.0), which is a worthwhile hit imo.

Reply:
I finally added it in c0fb984693dd09ccdfc840e0e8271ebbae19e225. My main concern was not about its size but that we don't bundle dependencies (so they can be shared between blocks/plugins), that means every new dependency we add produces another server request.

Reply:
> we don't bundle dependencies (so they can be shared between blocks/plugins), that means every new dependency we add produces another server request.

Ya that is one downside.  However I'm not too worried about additional server request as long as:

- it's non-blocking (or _minimal_ blocking)
- it's small in size.